### DISCUSSION TITLE
Typescript smart form implementation. Form doesn't triggers `validate`

### DISCUSSION THREAD

**Discussion by yogrr** [react-hook-form/react-hook-form#11155] [Q&A]
Found code example [at](https://react-hook-form.com/ts#UseFormReturn)

Forked a [sandbox](https://codesandbox.io/s/react-hook-form-useformmethods-forked-586xh9?file=/src/App.tsx)(original is buggy and contains errors)

I want to use custom validation callbacks at each form input element, so I did provide options in `register`, but `validate` is triggered only at submition even `useForm` is configured with
```TypeScript
{ reValidateMode: "onChange" }
```

The question is how to start `validate` on `onChange` event in that case described in sandbox

**Comment by craigmcc** [comment]
I use { mode: "onBlur" } in pretty much all of my RHF forms, and it triggers the individual validation rules immediately when the user tries to go to the next field.  Have you tried that?

**Reply by yogrr** [reply]
"onBlur" doesn't work too

I found out that type `InputProps` is based on html's attributes that have `onChane` and `onBlur` like `ReturnType` of `register` callback, so may be them are overlaping each other, even ommiting these props with `Omit` doesn't excluding `onChane` and `onBlur` from `Input` component
Is it possible that some TS tricks are happening there, that I haven't noticed yet?

**Comment by yogrr** [comment]
[Stackoverflow](https://stackoverflow.com/questions/77435359/react-hook-form-revalidating-doesnt-start)
